lib/zlib.js

```diff
@@ -310,13 +335,19 @@ Object.defineProperty(Zlib.prototype, '_closed', {
 });
 
 Zlib.prototype.params = function params(level, strategy, callback) {
-  if (level < Z_MIN_LEVEL || level > Z_MAX_LEVEL)
-    throw new errors.RangeError('ERR_INVALID_ARG_VALUE', 'level', level);
-
-  if (strategy !== undefined &&
-      (strategy < Z_DEFAULT_STRATEGY || strategy > Z_FIXED ||
-       !Number.isFinite(strategy))) {
-    throw new errors.TypeError('ERR_INVALID_ARG_VALUE', 'strategy', strategy);
+  if (validateFiniteNumber(level, 'level')) {
+    if (level < Z_MIN_LEVEL || level > Z_MAX_LEVEL) {
```

Comment:
The statements could be combined. So it would only say:

```js
if (validateFiniteNumber(level, 'level') && (level < ... || level > ...)) {
  // ...
}
```

The same for the one below.

Reply:
@BridgeAR That does not work because a valid value would be set overwritten to the default in the else block.

Reply:
@BridgeAR Ah..right, this is not one of the blocks that sets default values. Nevermind.